services:
  myapp:
    image: myapp:stable
    container_name: myapp
    restart: always
    environment:
      - AUTH__EMAIL__HOST=smtp.yandex.ru
      - AUTH__EMAIL__PORT=465
      - AUTH__EMAIL__USERNAME=user
      - AUTH__EMAIL__PASSWORD=password
      - AUTH__EMAIL__SENDER=user
      - AUTH__RABBITMQ__HOST=rabbitmq
      - AUTH__TEST_RABBITMQ__HOST=rabbitmq
      - AUTH__DB__HOST=database
      - AUTH__TEST_DB__HOST=database
      - AUTH__TEST_MODE=True
      - AUTH__HOST=localhost
    command: python -m pytest
    networks:
      - myappnet
    depends_on:
      rabbitmq:
        condition: service_healthy

  rabbitmq:
    container_name: rabbitmq
    hostname: rabbitmq
    image: rabbitmq:4.1.3-management-alpine
    restart: always
    networks:
      - myappnet
    environment:
      - RABBITMQ_DEFAULT_USER=admin
      - RABBITMQ_DEFAULT_PASS=admin
      - RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS=-rabbitmq log_levels [{connection,error},{default,error}] disk_free_limit 2147483648
    ports:
      - 15672:15672
    healthcheck:
      test: rabbitmq-diagnostics -q ping
      interval: 30s
      timeout: 30s
      retries: 3

networks:
  myappnet:
    driver: bridge